RandLAPACK/gpu_functions/rl_cuda_macros.hh

```diff
@@ -0,0 +1,28 @@
+#ifndef MACROS_HH
+#define MACROS_HH
+#include <cstdint>
+#include <iostream>
+
+#define RandLAPACK_CUDA_STATUS(_msg)                                                \
+    std::cout << "STATUS: " _msg << std::endl;
+
+#define RandLAPACK_CUDA_ERROR(_msg)                                                \
+    std::cerr << "ERROR: " << __FILE__ << ":" << __LINE__ << std::endl  \
+        << "" _msg << std::endl;
+
+#define RandLAPACK_CUDA_FATAL_ERROR(_msg)                                          \
+    std::cerr << "ERROR: " << __FILE__ << ":" << __LINE__ << std::endl  \
+        << "" _msg << std::endl;                                        \
+    abort();
+
+#define check_cuda_error                                                \
+{                                                                       \
+    cudaDeviceSynchronize();                                            \
+    cudaError_t ierr = cudaGetLastError();                              \
+    if (ierr != cudaSuccess)                                            \
+    {                                                                   \
+        RandLAPACK_CUDA_ERROR("CUDA error : " << cudaGetErrorString( ierr ));      \
+        abort();                                                        \
+    }                                                                   \
+}
```

Comment:
I think your life would improve if you used this macro in the other file.

Reply:
I want meaningful error messages.

Reply:
I'll probably open an issue on this.